pkg/instrumentation/annotationmutator.go

```diff
@@ -9,39 +9,30 @@ import (
 
 // AnnotationMutation is used to modify an annotation map.
 type AnnotationMutation interface {
-	// Mutate attempts to modify the annotations map. Returns whether the function changed the annotations.
-	Mutate(annotations map[string]string) bool
+	// Mutate attempts to modify the annotations map. Returns the mutated annotations if any and whether the function
+	// changed the annotations.
+	Mutate(annotations map[string]string) (map[string]string, bool)
 }
 
 type insertAnnotationMutation struct {
 	insert map[string]string
 }
 
-func (m *insertAnnotationMutation) Mutate(annotations map[string]string) bool {
-	if !m.shouldMutate(annotations) {
-		return false
-	}
+func (m *insertAnnotationMutation) Mutate(annotations map[string]string) (map[string]string, bool) {
 	var mutated bool
+	mutatedAnnotations := make(map[string]string)
 	for key, value := range m.insert {
 		if _, ok := annotations[key]; !ok {
 			annotations[key] = value
+			mutatedAnnotations[key] = value
 			mutated = true
 		}
 	}
-	return mutated
+	return mutatedAnnotations, mutated
```

Comment:
nit: Could remove the mutated bool or could change it to not return a bool at all since the presence of entries in the map would indicate a change

```suggestion
	return mutatedAnnotations, len(mutatedAnnotations) != 0
```